libbpf-tools/hardirqs.c

```diff
@@ -77,7 +81,16 @@ static error_t parse_arg(int key, char *arg, struct argp_state *state)
 	case 'C':
 		env.count = true;
 		break;
-	case 'c':
+        case 's':
+                errno = 0;
+                long c = strtol(arg, NULL, 10);
+                if (errno || c < 0) {
+		        fprintf(stderr, "invalid cpu: %s\n", arg);
+                        argp_usage(state);
+                }
+                env.targ_cpu = (int)c;
```

Comment:
This is not right. if c = 0xf00000002ULL, (int)c will.be 2. Maybe instead of strtol, using atoi?